### DISCUSSION TITLE
pyright-langserver options

### DISCUSSION THREAD

**Discussion by MijikHna** [microsoft/pyright#7600] [General]
Is there a way to find out `pyright-langserver` options?
I have found options for `pyright` but it seems like those options doesn't work for `pyright-langserver`. I was not able to find any resources in internet. I have also watched into docs folder of the repo but it seems those docs are related to `pyright` usage (Maybe I have just missed `pyright-langserver` related docs)

**Comment by erictraut** [comment]
The interface to `pyright-langserver` is defined by the [language server protocol](https://microsoft.github.io/language-server-protocol/) (LSP).

**Reply by MijikHna** [reply]
what I mean are options which I can provide on start. For example here are options which I can provide to `pyright` on start
```
ζ pyright --help                                                                                                                                                                                                     
Usage: pyright [options] files...
  Options:
  ...
  -h,--help                          Show this help message
  -p,--project <FILE OR DIRECTORY>   Use the configuration file at this location
  --pythonplatform <PLATFORM>        Analyze for a specific platform (Darwin, Linux, Windows)
  --pythonpath <FILE>                Path to the Python interpreter
  --pythonversion <VERSION>          Analyze for a specific version (3.3, 3.4, etc.)
  --skipunannotated                  Skip analysis of functions with no type annotations
  --stats                            Print detailed performance stats
  -t,--typeshedpath <DIRECTORY>      Use typeshed type stubs at this location
  -v,--venvpath <DIRECTORY>          Directory that contains virtual environments
  ...
```
I am especially interested in `--venvpath` and `--pythonpath`
Some background to my use case. I am using `pyright` as LSP for neovim. My problem at the moment is that `pyright` grabs my global python installation, but I wish that `pyright` just uses `venv` python. I also have tried `pyrightconfig.json`. I also tried to use `--venvpath` and `--pythonpath` as options besides `--stdio` for `pyright-langserver`.

**Reply by erictraut** [reply]
`pyright-langserver` is a language server, not a CLI tool. Its interface and the options it accepts are defined by the language server protocol. It honors config files (either pyrightconfig.json or pyproject.toml). And it accepts [LSP settings](https://microsoft.github.io/pyright/#/settings) using the LSP interface. The setting "python.pythonPath" is probably the one you are looking for. I recommend against using the "python.venvPath" configuration mechanism because it is much less robust (see [this section](https://microsoft.github.io/pyright/#/import-resolution?id=configuring-your-python-environment) for details).

**Reply by MijikHna** [reply]
Thank you. I will have a closer look into LSP Settings documentation

**Reply by wrf1998** [reply]
I had the same problem:
I have to start command for the node/node_modules/pyright/dist/pyright - langserver. Js - the stdio
Then I added the pyrightconfig.json configuration file in /node_modules/pyright/dist/ and it didn't work

<img width="1810" alt="image" src="https://github.com/microsoft/pyright/assets/33780900/f4e2864a-4c69-45b2-8b7f-134a6d7a96ea">

<img width="1882" alt="image" src="https://github.com/microsoft/pyright/assets/33780900/bf958224-e146-4432-b8a6-c3ab35cefc80">

**Reply by MijikHna** [reply]
@wrf1998  I guess you need to place `pyrightconfig.py` in the root of you project folder not in node_modues folders https://github.com/microsoft/pyright/blob/main/docs/configuration.md
